### PR TITLE
dbeaver/pro#2552 [TE] deleted profiles appear again after app launch

### PR DESCRIPTION
**Pull request by serjiokov**
Api fixed for store entire list of profiles

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceRegistry.java [hunk 1]
```diff
@@ -510,6 +511,15 @@ public void updateAuthProfile(DBAAuthProfile profile) {
         }
     }
 
+    @Override
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceRegistry.java:521]: In fact it is not "add". It is setAuthProfiles

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/app/DBPDataSourceRegistry.java [hunk 2]
```diff
@@ -128,6 +128,9 @@ public interface DBPDataSourceRegistry extends DBPObject, DBPSecretHolder {
     @NotNull
     List<DBAAuthProfile> getApplicableAuthProfiles(@Nullable DBPDriver driver);
     void updateAuthProfile(DBAAuthProfile profile);
+
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/app/DBPDataSourceRegistry.java]: Javadoc

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceRegistry.java (hunk 1 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceRegistry.java:521]: Done.

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/app/DBPDataSourceRegistry.java (hunk 2 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/app/DBPDataSourceRegistry.java]: Done.

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/app/DBPDataSourceRegistry.java [hunk 3]
```diff
@@ -128,6 +128,14 @@ public interface DBPDataSourceRegistry extends DBPObject, DBPSecretHolder {
...
+    /**
+     * Set collection of profiles.
+     *
+     * @param profiles - profile collection
+     */
+    void setAuthProfiles(Collection<DBAAuthProfile> profiles);
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/app/DBPDataSourceRegistry.java]: Please specify annotations

File: plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceRegistry.java [hunk 4]
```diff
@@ -510,6 +511,20 @@ public void updateAuthProfile(DBAAuthProfile profile) {
         }
     }
 
+    /**
+     * Set new collection of profiles.
+     *
+     * @param profiles - profile collection
+     */
+    @Override
+    public void setAuthProfiles(Collection<DBAAuthProfile> profiles) {
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceRegistry.java]: Same here

**Review by ShadelessFox — APPROVED** [review]

**Review by E1izabeth — APPROVED** [review]

**Pull request merged**